test/minikube/minikube_base_tde_test.go

```diff
@@ -0,0 +1,225 @@
+	if os.Getenv("NUODB_DEV") != "true" {
+		t.Skip("'tde_monitor' service is not supported in released versions")
+	}
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	options := helm.Options{
+		SetValues: map[string]string{
+			"admin.tde.secrets.demo":                "demo-tde-secret",
+			"database.name":                         "demo",
+			"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"admin.tde.storagePasswordsDir":         "/etc/nuodb/encryption",
+		},
+	}
+	randomSuffix := strings.ToLower(random.UniqueId())
+	namespaceName := fmt.Sprintf("admin-cold-start-with-tde-%s", randomSuffix)
+	testlib.CreateNamespace(t, namespaceName)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	options.KubectlOptions = kubectlOptions
+
+	password := strings.ToLower(random.UniqueId())
+	applyStoragePasswordSecret(t, namespaceName, options.SetValues["admin.tde.secrets.demo"], []string{password})
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+	helmChartReleaseName, _ := testlib.StartAdmin(t, &options, 1, namespaceName)
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", helmChartReleaseName)
+	testlib.Await(t, func() bool {
+		return testlib.GetStringOccurrenceInLog(t, namespaceName, admin0,
+			"Successfully updated storage passwords for dbName=demo", &corev1.PodLogOptions{}) >= 1
+	}, 30*time.Second)
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+	databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &options)
+	// Enable TDE on database level
+	output, _ := testlib.RunSQL(t, namespaceName, admin0, "demo", "alter database change encryption type AES128")
+	assert.False(t, strings.Contains(output, "Error"), "Failed to enable TDE: %s", output)
+
+	// Restarting the only admin pod will mean that all storage passwords
+	// will be "forgoten" by the admin
+	adminPod := testlib.GetPod(t, namespaceName, admin0)
+	testlib.DeletePod(t, namespaceName, "pod/"+admin0)
+	testlib.AwaitPodObjectRecreated(t, namespaceName, adminPod, 30*time.Second)
+	testlib.AwaitPodUp(t, namespaceName, admin0, 300*time.Second)
+	testlib.AwaitAdminFullyConnected(t, namespaceName, admin0, 1)
+	testlib.Await(t, func() bool {
+		return testlib.GetStringOccurrenceInLog(t, namespaceName, admin0,
+			"Successfully updated storage passwords for dbName=demo", &corev1.PodLogOptions{}) >= 1
+	}, 30*time.Second)
+
+	// verify that restarting an SM pod will succeed to start encrypted archive
+	opt := testlib.GetExtractedOptions(&options)
+	smPodNameTemplate := fmt.Sprintf("sm-%s-nuodb-%s-%s", databaseChartName, opt.ClusterName, opt.DbName)
+	smPodName0 := testlib.GetPodName(t, namespaceName, smPodNameTemplate)
+	testlib.DeletePod(t, namespaceName, "pod/"+smPodName0)
```

Comment:
I wish we had the ability to start multiple SMs with the CE license. An SM restart would not take down the whole database...

Reply:
Why would it take down the database? Only the SM that exceeds the limit on number of SMs would be rejected, but the original one would remain running.

Reply:
In this test, we have 1SM and 1TE database and this is why it will take the database down once we restart the SM.
I don't see a point in using multiple SMs in this case because we just want to test SM restart will succeed. If for some reason the password is not set or it's incorrect, the SM will fail to start with an error because it won't be able to decrypt the master password in the archive.